lib/auth.js

```diff
@@ -49,27 +38,61 @@ async function auth(getCredentials = ghauth) {
     process.stderr.write(`Could not get token: ${e.message}${EOL}`);
     process.exit(1);
   }
+  return credentials;
+}
 
-  const json = JSON.stringify({
-    username: credentials.user,
-    token: credentials.token
-  }, null, '  ');
-  await writeFile(authFile, json, { mode:
-    0o600 /* owner read/write */
-  });
-
-  return auth(getCredentials);
+function encode(name, token) {
+  return Buffer.from(`${name}:${token}`).toString('base64');
 }
 
-function lazy(fn) {
-  let cachedValue;
-  return function(...args) {
-    if (cachedValue !== undefined) {
-      return cachedValue;
+// TODO: support jenkins only...or not necessary?
+async function auth(
+  options = { github: true },
+  githubAuth = ghauth) {
+  const result = {};
+  if (options.github) {
+    let username;
+    let token;
+    try {
+      ({ username, token } = getMergedConfig());
+    } catch (e) {
+      // Ignore error and prompt
     }
-    cachedValue = fn(...args);
-    return cachedValue;
-  };
+
+    if (!username || !token) {
+      process.stdout.write(
+        'If this is your first time running this command, ' +
+        'follow the instructions to create an access token' +
+        '. If you prefer to create it yourself on Github, ' +
+        'see https://github.com/nodejs/node-core-utils/blob/master/README.md.' +
+        EOL);
+      const credentials = await tryCreateGitHubToken(githubAuth);
+      username = credentials.user;
+      token = credentials.token;
+      const json = JSON.stringify({ username, token }, null, 2);
+      fs.writeFileSync(getNcurcPath(), json, {
+        mode: 0o600 /* owner read/write */
+      });
+      // Try again reading the file
+      ({ username, token } = getMergedConfig());
+    }
+    check(username, token);
+    result.github = encode(username, token);
+  }
+
+  if (options.jenkins) {
+    let { username, jenkins_token } = getMergedConfig();
+    if (!username || !jenkins_token) {
+      process.stdout.write(
+        'Get your Jenkins API token in https://ci.nodejs.org/me/configure ' +
```

Comment:
I have not added any documentation on how to get the Jenkins API token but it's pretty obvious if you follow this error message. It's only used when you use `ncu-ci` anyway.